ducktape/command_line/parse_args.py

```diff
@@ -128,6 +128,19 @@ def config_file_to_args_list(config_file):
     return list(itertools.chain(*[line.split() for line in config_lines]))
 
 
+def parse_non_default_args(parser: argparse.ArgumentParser, defaults: dict, args: list) -> dict:
```

Comment:
why do you need to remove the defaults?

Reply:
otherwise the merging doesn't work, as the defaults will override the pervious files contributions. (the parser will always include the defaults when parsing the args)